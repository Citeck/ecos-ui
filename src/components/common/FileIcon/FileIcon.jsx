import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import formats from './font/icons/high-contrast/catalog';
import './font/file-icon-high-contrast.css';
import './FileIcon.scss';

const FileIcon = ({ format, className }) => {
  let isCorrectFormat = false;
  let lowerCaseFormat = format || '';
  if (format && typeof format.toLowerCase === 'function') {
    lowerCaseFormat = format.toLowerCase();
    isCorrectFormat = Array.from(formats).includes(lowerCaseFormat);
  }

  return (
    <span
      className={classNames('ecos-file-icon', 'fiv-hct', className, {
        [`fiv-icon-${lowerCaseFormat}`]: isCorrectFormat,
        'fiv-icon-blank': !isCorrectFormat || lowerCaseFormat === 'blank'
      })}
    />
  );
};

FileIcon.propTypes = {
  className: PropTypes.string,
  format: PropTypes.oneOf(Array.from(formats))
};

export default FileIcon;
