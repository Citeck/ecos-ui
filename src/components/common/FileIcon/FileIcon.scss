.ecos-file-icon {
  &:before {
    content: '';
  }
}
