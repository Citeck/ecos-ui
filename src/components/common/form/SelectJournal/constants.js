import { DataTypes, DisplayModes } from '../../../../forms/components/custom/selectJournal/constants';

export { DataTypes, DisplayModes };

export const Labels = {
  NO_JOURNAL_ID_ERROR: 'select-journal.error.no-journal-id',
  NO_JOURNAL_CONFIG_ERROR: 'select-journal.error.no-journal-config',
  DEFAULT_TITLE: 'select-journal.select-modal.title',
  FILTER_BUTTON: 'select-journal.select-modal.filter-button',
  CANCEL_BUTTON: 'select-journal.select-modal.cancel-button',
  SAVE_BUTTON: 'select-journal.select-modal.ok-button',
  MSG_WHOLE_SELECTION: 'select-journal.msg.whole-selection-used',
  PLACEHOLDER: 'select-journal.placeholder',
  INPUT_BTN_ADD: 'select-journal.button.add',
  INPUT_BTN_CHANGE: 'select-journal.button.change',
  INPUT_BTN_SELECT: 'select-journal.button.select',
  SELECTED_LABEL: 'select-journal.msg.selected-records'
};
