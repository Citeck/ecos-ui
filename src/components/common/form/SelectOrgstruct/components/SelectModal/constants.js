export const TAB_BY_LEVELS = 'TAB_BY_LEVELS';
export const TAB_ALL_USERS = 'TAB_ALL_USERS';
export const TAB_ONLY_SELECTED = 'TAB_ONLY_SELECTED';
