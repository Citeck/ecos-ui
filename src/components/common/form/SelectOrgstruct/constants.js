export const TAB_BY_LEVELS = 'TAB_BY_LEVELS';
export const TAB_ALL_USERS = 'TAB_ALL_USERS';
export const TAB_ONLY_SELECTED = 'TAB_ONLY_SELECTED';

export const ALL_USERS_GROUP_SHORT_NAME = 'all';

export const AUTHORITY_TYPE_GROUP = 'GROUP';
export const AUTHORITY_TYPE_USER = 'USER';

export const GroupTypes = {
  ROLE: 'ROLE',
  BRANCH: 'BRANCH',
  GROUP: 'GROUP'
};

export const ViewModes = {
  DEFAULT: 'default',
  LINE_SEPARATED: 'line-separated'
};
