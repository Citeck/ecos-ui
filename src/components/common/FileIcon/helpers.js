import formats from './font/icons/high-contrast/catalog';
import { blankFormat } from './constants';

export function detectFormat(fileName) {
  try {
    if (typeof fileName !== 'string') {
      return blankFormat;
    }

    const parts = fileName.split('.');
    if (parts.length < 2) {
      return blankFormat;
    }

    const lastPart = parts[parts.length - 1].toLowerCase();
    if (Array.from(formats).includes(lastPart)) {
      return lastPart;
    }

    return blankFormat;
  } catch (e) {
    return blankFormat;
  }
}
