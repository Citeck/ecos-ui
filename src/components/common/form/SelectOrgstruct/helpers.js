import isNil from 'lodash/isNil';
import isUndefined from 'lodash/isUndefined';

import { SourcesId } from '../../../../constants';
import { AUTHORITY_TYPE_USER, AUTHORITY_TYPE_GROUP } from './constants';

export const getGroupName = str => str.replace(`${AUTHORITY_TYPE_GROUP}_`, '');
export const getGroupRef = str => `${SourcesId.GROUP}@${str}`;
export const getPersonRef = str => `${SourcesId.PERSON}@${str}`;
export const getAuthRef = str => str.replace(`${SourcesId.GROUP}@`, `${AUTHORITY_TYPE_GROUP}_`).replace(`${SourcesId.PERSON}@`, '');

export function handleResponse(result) {
  if (!Array.isArray(result)) {
    result = [result];
  }

  return result.map(item => ({
    id: item.nodeRef,
    label: item.displayName,
    extraLabel: item.authorityType === AUTHORITY_TYPE_USER ? item.fullName : null,
    hasChildren: !isNil(item.groupType),
    isLoaded: isUndefined(item.isLoaded) ? false : item.isLoaded,
    isOpen: isUndefined(item.isOpen) ? false : item.isOpen,
    attributes: item
  }));
}

export function unionWithPrevious(newItems, oldItems) {
  return newItems.map(newItem => {
    const prevItem = oldItems.find(i => i.id === newItem.id);

    if (!prevItem) {
      return newItem;
    }

    newItem.isLoaded = prevItem.isLoaded;
    newItem.isOpen = prevItem.isOpen;
    newItem.isSelected = prevItem.isSelected;
    newItem.hasChildren = prevItem.isOpen ? prevItem.hasChildren : newItem.hasChildren;

    return newItem;
  });
}

export function prepareParentId(item, parentId) {
  return {
    ...item,
    parentId
  };
}

export function prepareSelected(selectedItem) {
  return {
    ...selectedItem,
    hasChildren: false,
    isSelected: true,
    parentId: undefined
  };
}

export function converterUserList(source) {
  return source.map(item => ({
    id: item.id,
    label: item.displayName,
    extraLabel: item.fullName,
    isPersonDisabled: !!item.isPersonDisabled,
    attributes: item
  }));
}

export function renderUsernameString(str, replacements) {
  const regex = /\${[^{]+}/g;

  function interpolate(template, variables, fallback) {
    return template.replace(regex, match => {
      const path = match.slice(2, -1).trim();

      return getObjPath(path, variables, fallback);
    });
  }

  function getObjPath(path, obj, fallback = '') {
    return path.split('.').reduce((res, key) => res[key] || fallback, obj);
  }

  return interpolate(str, replacements);
}
