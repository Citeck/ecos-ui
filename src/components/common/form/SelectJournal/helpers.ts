import get from 'lodash/get';

import { URL as Urls } from '@/constants';
import { createDocumentUrl, getLinkWithWs, getSelectedValueLink } from '@/helpers/urls';

export const getFormatter = (linkFormatter: string): (() => string) | null => {
  let formatterFunc = null;

  if (linkFormatter) {
    try {
      formatterFunc = eval(`(${linkFormatter})`);
    } catch (err) {
      console.error('[SelectJournal] Error when parsing the link Formatter:', err);
    }
  }

  return formatterFunc;
};

export const getFormattedLink = ({
  item,
  formatterFunc
}: {
  item: {
    locatedWorkspaceId?: string;
    disp: string;
    id: string;
    canEdit: boolean;
  };
  formatterFunc?: (link: string) => string;
}): string => {
  let link = getSelectedValueLink(item);
  const workspaceId = get(item, 'locatedWorkspaceId');

  if (formatterFunc) {
    try {
      link = formatterFunc(link);
    } catch (err) {
      console.error('[SelectJournal] Error in linkFormatter method:', err);
    }
  } else {
    link = createDocumentUrl(link);
  }

  if (!!link && !link.includes(Urls.DASHBOARD) && link.includes('@')) {
    link = createDocumentUrl(link);

    if (!!workspaceId) {
      link = getLinkWithWs(link, workspaceId);
    }
  }

  return link;
};
