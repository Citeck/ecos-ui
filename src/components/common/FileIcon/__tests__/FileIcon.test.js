import { render } from '@testing-library/react';
import React from 'react';

import FileIcon from '../FileIcon';
import { blankFormat } from '../constants';
import { detectFormat } from '../helpers';

console.error = jest.fn();

const mainFormats = ['doc', 'docx', 'xls', 'xlsx', 'ppt', 'pptx', 'pdf', 'jpg', 'jpeg', 'bmp', 'png', 'gif', 'tif', 'odf', 'msg'];

describe('FileIcon tests', () => {
  describe('<FileIcon />', () => {
    it('should render FileIcon component', () => {
      const component = render(<FileIcon />);
      expect(component.asFragment(<FileIcon />)).toMatchSnapshot();
    });

    it('should apply className', () => {
      const { container } = render(<FileIcon className="testClass" />);
      expect(container.getElementsByClassName('testClass')).toHaveLength(1);
    });

    describe('should display default icon when file format is invalid', () => {
      const inputs = [undefined, null, true, 1, 'unknown'];

      inputs.forEach(format => {
        it(`case ${format + ''}`, () => {
          const { container } = render(<FileIcon format={format} />);
          expect(container.getElementsByClassName('fiv-icon-blank')).toHaveLength(1);
        });
      });
    });

    describe('should display an icon corresponding to the file format', () => {
      mainFormats.forEach(format => {
        it(`case ${format + ''}`, () => {
          const { container } = render(<FileIcon format={format} />);
          expect(container.getElementsByClassName(`fiv-icon-${format}`)).toHaveLength(1);
        });
      });
    });
  });

  describe('detectFormat helper', () => {
    describe('should return format if filename contains correct extension', () => {
      const testCases = mainFormats.map(format => ({ input: `file.${format}`, output: format }));

      testCases.forEach(testCase => {
        it(`case ${testCase.input}`, () => {
          expect(detectFormat(testCase.input)).toBe(testCase.output);
        });
      });
    });

    describe('should return default format if filename is not a string', () => {
      const inputs = [undefined, null, true, 1, [], {}];
      inputs.forEach(fileName => {
        it(`case ${JSON.stringify(fileName)}`, () => {
          expect(detectFormat(fileName)).toBe(blankFormat);
        });
      });
    });

    it('should return default format if no file extension', () => {
      expect(detectFormat('some file name')).toBe(blankFormat);
    });

    it('should return default format when unknown file extension', () => {
      expect(detectFormat('some file name.unknown')).toBe(blankFormat);
    });

    it('should return format in lower case', () => {
      expect(detectFormat('File.JPG')).toBe('jpg');
    });
  });
});
