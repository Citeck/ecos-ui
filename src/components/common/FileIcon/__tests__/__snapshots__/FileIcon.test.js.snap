// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileIcon tests <FileIcon /> should render FileIcon component 1`] = `
<DocumentFragment>
  <span
    class="ecos-file-icon fiv-hct fiv-icon-blank"
  />
</DocumentFragment>
`;
